Recurring releases of insomnia app (INS-1429)

filfreire opened the pull request:
<!--
Please open an [Issue](https://github.com/kong/insomnia/issues/new) first to discuss new
features or non-trivial changes. Please provide as much detail as possible on the change as
possible including general description, implementation details, potential shortcomings, etc.

If this PR closes an issue, please mention "Closes #XX" where #XX is the issue number.

If this PR fixes a bug or regression, please make sure to add a test.
-->

## What is changed

This PR adds a new workflow that'll pre-bake artifacts when there's changes on PRs or on develop.

These artifacts can be used for all our major platforms to some exploratory/deep testing on a work in progress PR or latest develop without the need to build those locally.

Also changed:
- Update default electron-builder targets
- Allow setting a build ref. Changes the default version to `0.0.1-dev+unknown`, so that build/packaging can be easily ran locally without setting any environment variables.

## How to test

You should now be able to just run `npm run app-package` without specifying git SHAs or Github references environment variables.

You should be able to run `smoke-test` related package command without any issues locally.

You can also mimic what happens on the new Github Actions workflow, locally, running something like:
```shell
BUILD_REF='1234' npm run app-package
```

This `BUILD_REF` env variable is for now intended to be used only in Github actions/ CI. 

It serves to add info to the version field on `About Insomnia`  menu, in order to help folks be sure of the build they are testing. 

In the case of Github Actions artifacts, it can for example be git's short SHA, like `BUILD_REF='$(git rev-parse --short HEAD)'`:

![Screenshot 2022-02-09 at 16 16 37](https://user-images.githubusercontent.com/11976836/153243025-0d224fd9-b36c-45ed-9790-5c7b9a3d7504.png)

cc @wongstein @wdawson - after this gets merged, it should enable us to easily download builds for a PR or latest develop if we want to test out something - effectively doing what we initially planned with nightlies.

DMarby committed:
Update default electron-builder targetgs

DMarby committed:
Allow setting a build ref

Changes the default version to `0.0.1-dev+unknown`, so that build/packaging can be easily ran locally without setting any environment variables.

filfreire committed:
Adds recurring release workflow

New workflow that'll pre-bake artifacts when there's changes on PRs or on develop.

Artifacts can be used for all platforms to test out a PR or latest develop without the need to build locally.

Co-Authored-By: David Marby <david@dmarby.se>

filfreire committed:
Make recurring flow independent from Test flow

Co-Authored-By: David Marby <david@dmarby.se>

filfreire committed:
Use setup-node@v2

Co-Authored-By: David Marby <david@dmarby.se>

filfreire committed:
Merge branch 'develop' into feature/ins-1429-recurring-releases-of-insomnia-app

DMarby commented on .github/workflows/release-recurring.yml:
```diff
@@ -0,0 +1,49 @@
+         - os: "windows-latest"
+           build-targets: "portable"
+         - os: "ubuntu-latest"
+           build-targets: ".tar.gz"
```
```suggestion
           build-targets: "tar.gz"
```

DMarby reviewed (approved):
🎉

filfreire committed:
Update .github/workflows/release-recurring.yml

Co-authored-by: David Marby <david@dmarby.se>

filfreire committed:
It's always the single quotes 🙏 😚

DMarby committed:
Use bash shell

DMarby committed:
Disable fail-fast

filfreire committed:
Add cancel-in-progress setting to recurring releases

If someone pushes a new commit, triggering a new recurring job on a PR or develop, we cancel ongoing workflow.

Co-Authored-By: David Marby <david@dmarby.se>

filfreire committed:
Change concurrency setting to workflow level

Co-Authored-By: David Marby <david@dmarby.se>

filfreire committed:
Merge branch 'develop' into feature/ins-1429-recurring-releases-of-insomnia-app
